src/main/scala/fr/acinq/bitcoin/scalacompat/Musig2.scala

```diff
@@ -1,33 +1,51 @@
 package fr.acinq.bitcoin.scalacompat
 
-import fr.acinq.bitcoin.ScriptTree
-import fr.acinq.bitcoin.crypto.musig2.{IndividualNonce, SecretNonce}
+import fr.acinq.bitcoin.crypto.musig2
 import fr.acinq.bitcoin.scalacompat.Crypto.{PrivateKey, PublicKey, XonlyPublicKey}
 import fr.acinq.bitcoin.scalacompat.KotlinUtils._
+import fr.acinq.secp256k1.Secp256k1
+import scodec.bits.ByteVector
 
 import scala.jdk.CollectionConverters.SeqHasAsJava
 
 object Musig2 {
 
+  /**
+   * Musig2 secret nonce, that should be treated as a private opaque blob.
+   * This nonce must never be persisted or reused across signing sessions.
+   */
+  case class SecretNonce(inner: musig2.SecretNonce)
+
+  /**
+   * Musig2 public nonce, that must be shared with other participants in the signing session.
+   * It contains two elliptic curve points, but should be treated as an opaque blob.
+   */
+  case class IndividualNonce(data: ByteVector) {
+    require(data.size == Secp256k1.MUSIG2_PUBLIC_NONCE_SIZE, "invalid musig2 public nonce size")
+  }
+
+  /** A locally-generated nonce, for which both the secret and public parts are known. */
+  case class LocalNonce(secret: SecretNonce, public: IndividualNonce)
```

Comment:
nit: 
```suggestion
  case class LocalNonce(secretNonce: SecretNonce, publicNonce: IndividualNonce)
```
would be easier to read  (and also to search  for especially for `public` which is a keyword in Kotlin and Scala)